search-service/src/test/kotlin/com/egm/stellio/search/service/EntityEventListenerServiceTest.kt

```diff
@@ -197,8 +165,113 @@ class EntityEventListenerServiceTest {
                 }
             }
             """.trimIndent()
-        val content = prepareUpdateEventPayload(eventPayload)
 
+        assertAttributeInstanceWithDatasetIdCreation(prepareAttributeReplaceEventPayload(eventPayload))
+    }
+
+    @Test
+    fun `it should ignore an attribute instance without observedAt information`() {
+        val eventPayload =
+            """
+            {
+                \"totalDissolvedSolids\":{
+                    \"type\":\"Property\",
+                    \"value\":33869
+                }
+            }
+            """.trimIndent()
+        val content = prepareAttributeReplaceEventPayload(eventPayload)
+
+        entityEventListenerService.processMessage(content)
+
+        verify {
+            temporalEntityAttributeService.getForEntityAndAttribute(any(), any()) wasNot Called
+        }
+    }
+
+    @Test
+    fun `it should create an attribute instance and update entity payload for attributeUpdate events`() {
+        val eventPayload =
+            """
+            {
+                \"type\":\"Property\",
+                \"value\":33869,
+                \"observedAt\":\"$observedAt\"
+            }
+            """.trimIndent()
+
+        assertAttributeInstanceCreation(prepareAttributeUpdateEventPayload(eventPayload))
+    }
+
+    @Test
+    fun `it should create an attribute instance with a datasetId for attributeUpdate events`() {
+        val eventPayload =
+            """
+            {
+                \"type\":\"Property\",
+                \"value\":\"some textual value\",
+                \"observedAt\":\"$observedAt\",
+                \"datasetId\": \"urn:ngsi-ld:Dataset:01234\"
+            }
+            """.trimIndent()
+
+        assertAttributeInstanceWithDatasetIdCreation(prepareAttributeUpdateEventPayload(eventPayload))
+    }
+
+    @Test
+    fun `it should ignore an attribute instance without observedAt information for attributeUpdate events`() {
+        val eventPayload =
+            """
+            {
+                \"type\":\"Property\",
+                \"value\":33869
+            }
+            """.trimIndent()
+        val content = prepareAttributeUpdateEventPayload(eventPayload)
+
+        entityEventListenerService.processMessage(content)
+
+        verify {
+            temporalEntityAttributeService.getForEntityAndAttribute(any(), any()) wasNot Called
+        }
+    }
+
+    private fun assertAttributeInstanceCreation(content: String) {
```

Comment:
this function name is a bit misleading because it not only asserts but also prepares the mocks and makes the call to the function under test. I would keep the concerns in separate functions (the code of the test functions will also be easier to read then).

Reply:
I've made some updates in the test class to illustrate my previous comment.